### ISSUE TITLE
Add support for Position setting to Template Parts for Sticky Headers

### ISSUE THREAD

**Issue by masperber** [WordPress/gutenberg#50617]
Apologies if this is a duplicate issue. I searched for a related issue and couldn't find one.

The Group Block has a very useful setting called Position which allows you to Sticky it to the top of the page. It would be great if Template Parts--specifically the Header Template Part--had support for this setting.

## What problem does this address?
Presently, if you want to make a template with a sticky header, you must encapsulate the header in a Group Block, and you cannot use the Header Template part.

## What is your proposed solution?
Add support for the Position > Sticky setting to Template Parts, or at least the Header Template part.

I also suggest that it would be useful to add a Position > Bottom setting to the Footer template part.

**Comment by andrewserong** [comment]
Thanks for opening up this issue! Unfortunately it isn't a simple one to implement because of the confusion that can be created when opening up styling controls directly in the template part itself. I had a go at this previously in https://github.com/WordPress/gutenberg/pull/47133 — the problem is that if you set the template part to sticky, it doesn't propagate across all instances of the template part, which can be confusing. And conversely, if we were to somehow get it to propagate by storing metadata somewhere, that also might not be the desired goal for the user.

The feedback on that PR was that it's better not to expose styling controls on template parts just yet (https://github.com/WordPress/gutenberg/pull/47133#issuecomment-1385285873), until there's a more flexible way of controlling style values at the template part level, that factors in the local vs global problem for template parts.

> Apologies if this is a duplicate issue. I searched for a related issue and couldn't find one.

I don't think there was a separate issue for this one, so thanks again for opening it. I'll link to it from the list of follow up tasks on the sticky position tracking issue: https://github.com/WordPress/gutenberg/issues/47043

**Comment by mxhassani** [comment]
Another case here: 8988663-zen

**Comment by bradhogan** [comment]
Would love to see this added.  I understand the concerns or difficulties, but it would be useful.

**Comment by youknowriad** [comment]
Maybe we can start by "local styles" like we used to have but clarify in the sidebar of the block (add a note or something)

**Comment by andrewserong** [comment]
One problem with starting by adding "local styles" is that users will still need to go and update each of their templates in order to have sticky headers across the site. Back when the sticky position support was added there wasn't widespread browser support for `:has()`. I wonder if it'd be better for us to include an opinionated style to handle this case? E.g. what's covered in this blog post: https://olliewp.com/fixing-the-sticky-header-bug-in-the-wordpress-site-editor/ (`header:has(>.is-position-sticky) {` or similar)

(thanks @cbravobernal for bringing this up, and to folks chatting about similar approaches in https://github.com/WordPress/gutenberg/issues/47043#issuecomment-3283788119)

**Comment by youknowriad** [comment]
@andrewserong Are you saying that all headers should be sticky? That doesn't seem like a good approach to me.

**Comment by youknowriad** [comment]
I think what you're saying is that if a template has a group that is sticky, it should be sticky too or something? That makes a little bit more sense. While this would work, how to systematize this. What happens if you use a pattern and not a template part, what if you had a template part within a template part... Is there a generic solution?

**Comment by andrewserong** [comment]
> I think what you're saying is that if a template has a group that is sticky, it should be sticky too or something? 

Yes, I believe that's the use case that's described in [the blog post](https://olliewp.com/fixing-the-sticky-header-bug-in-the-wordpress-site-editor/).

One of the clever things about the rule in that example is that it's `header:has(>.is-position-sticky) {` and not `.wp-block-template-part:has...` so it's a rule that applies to something that is semantically the header in the template. I think this is pretty neat because it side-steps the question of whether it's a template part or a pattern or where and how things are nested. It's simply a part / child relationship where any `header` element can effectively assume the sticky role of its child.

If we go down that path, I believe we'd need to be quite careful about how the rule is constructed for the following:
* Make sure it applies in situations where we can reasonably assume that the user has attempted to create a sticky header and:
* That the rule wouldn't adversely affect any intentional sticky behaviour that's used outside of a sticky header situation

`header:has(>.is-position-sticky) {` seems pretty good to me to cover that situation. I'm sure there are additional nuances to this kind of rule, though, so it could be good to try out a whole bunch of different potential headers / layouts to make sure any proposed rule really does work properly in a variety of use cases.

But overall, I don't mind this idea of the sticky position support bundling an opinionated rule for such a common use case.
